core/services/promreporter/prom_reporter.go

```diff
@@ -23,7 +24,7 @@ import (
 type (
 	promReporter struct {
 		services.StateMachine
-		db           *sql.DB
+		txStore      txmgr.TxStore
```

Comment:
I don't think we want to expose the TxStore directly to external components even if it's for reads. It'd make the work to add an in-memory layer trickier if services can use the methods that directly hit the DB. We'd want to use the TXM so we should expose these methods through the TXM interface. Although I don't think you can just pass the TXM directly here since we'd have an instance for each chain ID. I'll leave a suggestion on `application.go` on what I think you may need to do

Reply:
Ah, yes I missed this one.
Please add these new methods to TxManager interface, and directly call via that.
The txStore should be kept internal to the TXM.